quickwit/quickwit-ingest/src/ingest_v2/workbench.rs

```diff
@@ -161,6 +171,27 @@ impl IngestWorkbench {
         self.record_failure(open_shards_failure.subrequest_id, last_failure);
     }
 
+    pub fn record_get_or_create_open_shards_error(
+        &mut self,
+        subrequest_id: SubrequestId,
+        open_shard_error: &ControlPlaneError,
+    ) {
+        let last_failure = match open_shard_error {
+            ControlPlaneError::Internal(_) => SubworkbenchFailure::Internal,
+            ControlPlaneError::Timeout(_) => SubworkbenchFailure::ControlPlaneUnavailable,
+            ControlPlaneError::TooManyRequests => SubworkbenchFailure::ControlPlaneUnavailable,
+            ControlPlaneError::Unavailable(_) => SubworkbenchFailure::ControlPlaneUnavailable,
+            ControlPlaneError::Metastore(metastore_error) => match metastore_error {
+                MetastoreError::Timeout(_) => SubworkbenchFailure::ControlPlaneUnavailable,
+                MetastoreError::TooManyRequests => SubworkbenchFailure::ControlPlaneUnavailable,
+                MetastoreError::Unavailable(_) => SubworkbenchFailure::ControlPlaneUnavailable,
+                // TODO: are there other metastore errors that can be considered temporary?
+                _ => SubworkbenchFailure::Internal,
+            },
+        };
```

Comment:
This mapping determines which requests are going to be 500 or 503. In either case they will be retried internally (`is_pending()` returns true for both)